File: app/src/main/java/com/bkjcb/rqapplication/treatmentdefect/fragment/TreatmentDetailFragment.java
package com.bkjcb.rqapplication.treatmentdefect.fragment;

import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.fragment.BaseSimpleFragment;
import com.bkjcb.rqapplication.treatmentdefect.TreatmentDefectActivity;
import com.bkjcb.rqapplication.treatmentdefect.adapter.OrderListAdapter;
import com.bkjcb.rqapplication.treatmentdefect.adapter.SecurityCheckListAdapter;
import com.bkjcb.rqapplication.treatmentdefect.model.BottleResult;
import com.bkjcb.rqapplication.treatmentdefect.model.BottleSaleCheck;
import com.bkjcb.rqapplication.treatmentdefect.model.DefectTreatmentModel;
import com.bkjcb.rqapplication.retrofit.DataService;
import com.bkjcb.rqapplication.retrofit.NetworkApi;
import com.bkjcb.rqapplication.view.MyDialogViewHolder;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.hss01248.dialog.StyledDialog;
import com.hss01248.dialog.config.ConfigBean;

import java.util.List;

import butterknife.BindView;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by DengShuai on 2020/6/12.
 * Description :
 */
public class TreatmentDetailFragment extends BaseSimpleFragment {
    @BindView(R.id.info_distribution_list)
    RecyclerView mInfoDistributionList;
    @BindView(R.id.info_check_list)
    RecyclerView mInfoCheckList;
    @BindView(R.id.info_result)
    TextView mInfoResult;
    @BindView(R.id.info_accident_type)
    TextView mInfoAccidentType;
    @BindView(R.id.info_opinion)
    TextView mInfoOpinion;
    @BindView(R.id.info_type)
    TextView mInfoType;
    @BindView(R.id.info_station)
    TextView mInfoStation;
    @BindView(R.id.info_date)
    TextView mInfoDate;
    @BindView(R.id.info_year)
    TextView mInfoYear;
    @BindView(R.id.info_name)
    TextView mInfoName;
    @BindView(R.id.defect_detail)
    TextView mDefectDetail;
    private DefectTreatmentModel model;
    private SecurityCheckListAdapter checkAdapter;
    private OrderListAdapter orderAdapter;
    private ConfigBean configBean1;
    private ConfigBean configBean2;

    public void setModel(DefectTreatmentModel model) {
        this.model = model;
    }

    public static TreatmentDetailFragment newInstance(DefectTreatmentModel model) {
        TreatmentDetailFragment fragment = new TreatmentDetailFragment();
        fragment.setModel(model);
        return fragment;
    }

    @Override
    public void setResId() {
        resId = R.layout.fragment_treatment_detail;
    }

    @Override
    protected void initView() {
        StyledDialog.init(context);
        if (model != null) {
            mInfoResult.setText(model.getProcessTime());
            mInfoAccidentType.setText(model.getCasesType());
            mInfoOpinion.setText(model.getOpinions());
            mInfoType.setText(model.getUserCode());
            mInfoStation.setText(model.getUserName());
            mInfoYear.setText(model.getUserType() == 0 ? "居民" : "非居民");
            mInfoName.setText(model.getQu());
            mInfoDate.setText(model.getUserAddress());

            mInfoCheckList.setLayoutManager(new LinearLayoutManager(context));
            checkAdapter = new SecurityCheckListAdapter(R.layout.item_security_check_view);
            mInfoCheckList.setAdapter(checkAdapter);
            checkAdapter.bindToRecyclerView(mInfoCheckList);
            checkAdapter.showLoading();
            checkAdapter.setOnItemClickListener(new BaseQuickAdapter.OnItemClickListener() {
                @Override
                public void onItemClick(BaseQuickAdapter adapter, View view, int position) {
                    showDetail((BottleSaleCheck) adapter.getItem(position), 1);
                }
            });

            mInfoDistributionList.setLayoutManager(new LinearLayoutManager(context));
            orderAdapter = new OrderListAdapter(R.layout.item_order_view);
            mInfoDistributionList.setAdapter(orderAdapter);
            orderAdapter.bindToRecyclerView(mInfoDistributionList);
            orderAdapter.showLoading();
            orderAdapter.setOnItemClickListener(new BaseQuickAdapter.OnItemClickListener() {
                @Override
                public void onItemClick(BaseQuickAdapter adapter, View view, int position) {
                    showDetail((BottleSaleCheck) adapter.getItem(position), 0);
                }
            });
            if (model.getFlag() == 1) {
                showFinishBtn();
            }
        }
    }

    @Override
    protected void initData() {
        getBaseInfo();
    }

    private void getBaseInfo() {
        disposable = NetworkApi.getService(DataService.class)
                .getBottleData(model.getMbuGuid())
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<BottleResult>() {
                    @Override
                    public void accept(BottleResult result) throws Exception {
                        if (result.pushState == 200) {
                            setCheckData(result.getDatas().getBottleSaleChecks());
                            setOrderData(result.getDatas().getBottleSaleChecks());
                        } else {
                            showError();
                        }
                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        showError();
                    }
                });
    }

    private void setOrderData(List<BottleSaleCheck> list) {
        if (list != null && list.size() > 0) {
            orderAdapter.setNewData(list);
        } else {
            orderAdapter.showEmpty();
        }
    }

    private void setCheckData(List<BottleSaleCheck> list) {
        if (list != null && list.size() > 0) {
            checkAdapter.setNewData(list);
        } else {
            checkAdapter.showEmpty();
        }
    }

    private void showError() {
        checkAdapter.showError();
        orderAdapter.showError();
    }

    private void showFinishBtn() {
        mDefectDetail.setVisibility(View.VISIBLE);
        mDefectDetail.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                TreatmentDefectActivity.toActivity(getActivity(), model);
            }
        });
    }

    private void showDetail(BottleSaleCheck check, int type) {
        if (type == 0) {
            if (configBean1 == null) {
                MyDialogViewHolder holder1 = new MyDialogViewHolder(context, 0);
                configBean1 = StyledDialog.buildCustomInIos(holder1, null).setBtnText("确定");
            }
            ((MyDialogViewHolder) configBean1.customContentHolder).assingDatasAndEvents(context, check);
            configBean1.show();
        } else {
            if (configBean2 == null) {
                MyDialogViewHolder holder2 = new MyDialogViewHolder(context, 1);
                configBean2 = StyledDialog.buildCustomInIos(holder2, null).setBtnText("确定");
            }
            ((MyDialogViewHolder) configBean2.customContentHolder).assingDatasAndEvents(context, check);
            configBean2.show();

        }
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/treatmentdefect/fragment/TreatmentDetailFragment.kt
package com.bkjcb.rqapplication.treatmentdefect.fragment

import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import butterknife.BindView
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.fragment.BaseSimpleFragment
import com.bkjcb.rqapplication.retrofit.DataService
import com.bkjcb.rqapplication.retrofit.NetworkApi
import com.bkjcb.rqapplication.treatmentdefect.TreatmentDefectActivity
import com.bkjcb.rqapplication.treatmentdefect.adapter.OrderListAdapter
import com.bkjcb.rqapplication.treatmentdefect.adapter.SecurityCheckListAdapter
import com.bkjcb.rqapplication.treatmentdefect.model.BottleSaleCheck
import com.bkjcb.rqapplication.treatmentdefect.model.DefectTreatmentModel
import com.bkjcb.rqapplication.view.MyDialogViewHolder
import com.chad.library.adapter.base.BaseQuickAdapter
import com.hss01248.dialog.StyledDialog
import com.hss01248.dialog.config.ConfigBean
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.fragment_treatment_detail.*

/**
 * Created by DengShuai on 2020/6/12.
 * Description :
 */
class TreatmentDetailFragment : BaseSimpleFragment() {

    private var model: DefectTreatmentModel? = null
    private lateinit var checkAdapter: SecurityCheckListAdapter
    private lateinit var orderAdapter: OrderListAdapter
    private var configBean1: ConfigBean? = null
    private var configBean2: ConfigBean? = null
    fun setModel(model: DefectTreatmentModel?) {
        this.model = model
    }

    override fun initView() {
        StyledDialog.init(context)
        if (model != null) {
            info_result.text = model!!.processTime
            info_accident_type.text = model!!.casesType
            info_opinion.text = model!!.opinions
            info_type.text = model!!.userCode
            info_station.text = model!!.userName
            info_year.text = if (model!!.userType == 0) "居民" else "非居民"
            info_name.text = model!!.qu
            info_date.text = model!!.userAddress
            info_check_list.layoutManager = LinearLayoutManager(context)
            checkAdapter = SecurityCheckListAdapter(R.layout.item_security_check_view)
            info_check_list.adapter = checkAdapter
            checkAdapter.bindToRecyclerView(info_check_list)
            checkAdapter.showLoading()
            checkAdapter.onItemClickListener = BaseQuickAdapter.OnItemClickListener { adapter, view, position -> showDetail(adapter.getItem(position) as BottleSaleCheck?, 1) }
            info_distribution_list.layoutManager = LinearLayoutManager(context)
            orderAdapter = OrderListAdapter(R.layout.item_order_view)
            info_distribution_list.adapter = orderAdapter
            orderAdapter.bindToRecyclerView(info_distribution_list)
            orderAdapter.showLoading()
            orderAdapter.onItemClickListener = BaseQuickAdapter.OnItemClickListener { adapter, view, position -> showDetail(adapter.getItem(position) as BottleSaleCheck?, 0) }
            if (model!!.flag > 0) {
                showFinishBtn()
            }
        }
    }

    override fun initData() {
        obtainBaseInfo()
    }

    private fun obtainBaseInfo() {
            disposable = NetworkApi.getService(DataService::class.java)
                    .getBottleData(model!!.mbuGuid)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe({ result ->
                        if (result.pushState == 200) {
                            setCheckData(result.datas.bottleSaleChecks)
                            //setOrderData(result.datas.bottleSaleChecks)
                        } else {
                            showError()
                        }
                    }) { showError() }
        }

    private fun setOrderData(list: List<BottleSaleCheck>?) {
        if (list != null && list.isNotEmpty()) {
            orderAdapter.setNewData(list)
        } else {
            orderAdapter.showEmpty()
        }
    }

    private fun setCheckData(list: List<BottleSaleCheck>?) {
        if (list != null && list.isNotEmpty()) {
            checkAdapter.setNewData(list)
            orderAdapter.setNewData(list)
        } else {
            checkAdapter.showEmpty()
            orderAdapter.showEmpty()
        }
    }

    private fun showError() {
        checkAdapter.showError()
        orderAdapter.showError()
    }

    private fun showFinishBtn() {
        defect_detail.visibility = View.VISIBLE
        defect_detail.setOnClickListener { TreatmentDefectActivity.toActivity(activity, model) }
    }

    private fun showDetail(check: BottleSaleCheck?, type: Int) {
        if (type == 0) {
            if (configBean1 == null) {
                val holder1 = MyDialogViewHolder(context, 0)
                configBean1 = StyledDialog.buildCustomInIos(holder1, null).setBtnText("确定")
            }
            (configBean1!!.customContentHolder as MyDialogViewHolder).assingDatasAndEvents(context, check)
            configBean1!!.show()
        } else {
            if (configBean2 == null) {
                val holder2 = MyDialogViewHolder(context, 1)
                configBean2 = StyledDialog.buildCustomInIos(holder2, null).setBtnText("确定")
            }
            (configBean2!!.customContentHolder as MyDialogViewHolder).assingDatasAndEvents(context, check)
            configBean2!!.show()
        }
    }

    override fun initResID(): Int {
        return R.layout.fragment_treatment_detail
    }

    companion object {
        fun newInstance(model: DefectTreatmentModel?): TreatmentDetailFragment {
            val fragment = TreatmentDetailFragment()
            fragment.setModel(model)
            return fragment
        }
    }
}